### DISCUSSION TITLE
NavigationControl compass regression

### DISCUSSION THREAD

**Discussion by raboczi** [maplibre/maplibre-gl-js#5195] [General]
Sometime after v5.0.0-pre.7, there seems to have been a regression in the compass part of the NavigationControl. Here's a minimal way to reproduce the problem alongside an unpacked v5.0.0-pre.9 dist.zip:
```
<html lang="en">
  <head>
    <link rel="stylesheet" href="dist/maplibre-gl.css"/>
    <script src="dist/maplibre-gl.js"></script>
  </head>
  <body>
    <div id="container" style="height:100%; width:100%"></div>
    <script>
      const MAPTILER_KEY = 'get_your_own_OpIi9ZULNHzrESv6T2vL';
      var map = new maplibregl.Map({
        container: 'container',
        style: `https://api.maptiler.com/maps/basic-v2/style.json?key=${MAPTILER_KEY}`,
      })
      .addControl(new maplibregl.NavigationControl({
        showCompass: true
      }))
    </script>
  </body>
</html>
```
The map displays, but the navigation control won't be present. The browser error console shows "ReferenceError: Can't find variable: TouchEvent" at drag_move_state_manager.ts line 129.

If you change showCompass to false, the navigation control will display properly with the zoom in and zoom out controls.

**Comment by HarelM** [comment]
I've touched this code recently, but I tried to test it thoroughly on Chrome.
In which browser this doesn't work for you?
It might be that there's a need for "if -else if" in the following code:
https://github.com/maplibre/maplibre-gl-js/blob/main/src/ui/handler/drag_move_state_manager.ts

**Reply by raboczi** [reply]
Works on Chrome, Opera and Edge.
Fails on Firefox and Safari.
All of this is on MacOS 15.2.

**Reply by HarelM** [reply]
Ahh, that might explain it... Would you like to open a PR to solve this?

**Reply by HarelM** [reply]
This is probably related to this PR:
https://github.com/maplibre/maplibre-gl-js/pull/5104

**Reply by raboczi** [reply]
> Ahh, that might explain it... Would you like to open a PR to solve this?

I've never touched TypeScript before, so it'll take some time before I'm be able to raise a reasonable PR.

**Reply by raboczi** [reply]
PR at https://github.com/maplibre/maplibre-gl-js/pull/5205
